Examine System: hover text implementation and text visibility

Ryan089 opened the pull request:
## Summary

Implements hover text functionality for the Examine system, and miscellaneous system improvements.

Whenever the cursor is over an examinable object, the name of that object now appears at the bottom of the screen. Visibility of text-based Examine UI's now use an outline to help distinguish them in bright settings. SkinnedMeshRenderers are now correctly identified by the Examine system - previously it only recognised the model bind pose. Finally, I *think* that the SingleColourMaterial prefab no longer updates, which caused it to want to continually be committed to the repository in random PR's.

## Pictures/Videos (optional)

![Examine System visibility](https://user-images.githubusercontent.com/64360820/128702257-90445ee1-a5a8-44f0-8ca3-86d1c5641fe5.png)

## Known issues

While this adds more functionality, the trade-off is that the requirement to keep the examine camera always on (instead of simply when Shift is held down) means that this comes with a performance cost on GPU-bound systems.

## Fixes (optional)

Closes #747 
Closes #750

Ryan089 committed:
CompositeItemSelector.cs no longer records the mesh texture.

Upon review of the code, it was clear that the recorded texture was never actually used anywhere. Additionally, by allocating the texture to the singleColourMaterial, it was changing that file every time the Examine system was used. This caused the file to want to update within version control after every commit.

Ryan089 committed:
Created a hover text field on the Examine UI.

This field is configured to display the name of the object under the mouse cursor, even when the examine button (shift) is not held down. To accommodate this, all IExamineData are now required to have a GetName() method.

Ryan089 committed:
Examine now correctly identifies SkinnedMeshRenderers.

Previously, the SkinnedMeshRenderer mesh used by the Examine system was simply the model bind pose, which gave incorrect examine results. Now, the mesh is baked prior to rendering, ensuring that the Examine system returns a pixel-perfect match.

Ryan089 committed:
Added an outline to the Examine text.

cosmiccoincidence commented:
> While this adds more functionality, the trade-off is that the requirement to keep the examine camera always on (instead of simply when Shift is held down) means that this comes with a performance cost on GPU-bound systems.

Although this is a default feature in SS13, we could make it a toggleable setting so those users who do not like the added visual noise or may want to remove it to help their graphic performance can do so.

cosmiccoincidence commented:
I didn't make a build but I tested it in Unity and my hover-text does not seem to be correctly placed as shown in your image.

![image](https://user-images.githubusercontent.com/41941008/128919557-743f7975-61f1-411f-8e40-9f37bc92daab.png)

cosmiccoincidence commented:
Actually, I did notice 1 other thing...
The text outline seems to be on all text now, including the chat text. Is there a way to only apply this to the examine system?

Ryan089 committed:
Added a new material asset for Examine text font.

The new asset is required so that the Examine text can have an outline, while the original font can be used elsewhere (i.e. Chat system) without an outline.

Ryan089 committed:
Added the new examine font to the simple text and simple image UI's.

Ryan089 committed:
Turned off hover text when another examine UI element is in use.

Allowed setting in ExamineUI to turn off hover text completely.

Ryan089 committed:
Removed unnecessary matrix recalculation calls.

Ryan089 committed:
Examine font sizes changed.
- Hover text: 30pt
- Simple text (name): 30pt
- Simple text (description): 26pt
- Simple image: 30pt

Ryan089 committed:
Changed simple examine description to bold text.

cosmiccoincidence reviewed (changes requested):
- Hovertext is not centered above the hands inventory UI.

- The text outline should only apply to the examine system and not the text on the chat system.

- When the user detail-examines near the hovertext, both examines disappear... Only the hovertext should be disappearing.

Ryan089 committed:
Centred hover text. Removed outline from chat font.

cosmiccoincidence reviewed (approved):
I think there is more room for improvements but I'm calling this good for now. Good job.
